### ISSUE TITLE
Warning from ddt_load breaking -Werror

### ISSUE THREAD

**Issue by rincebrain** [openzfs/zfs#15905]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Debian
Distribution Version	| 11
Kernel Version	| 5.10.0-27-amd64
Architecture	| amd64
OpenZFS Version	| af4da5ccf28e1de6d4f6771a8d9d7cc578aba087

### Describe the problem you're observing
I'm guessing this is fallout from something in the recent DDT series, but I'm not sure why the CI wouldn't have screamed about it...

I was cutting a build to test something, and got this on vanilla git:
```
In file included from ./include/sys/spa.h:46,
                 from module/zfs/ddt.c:30:
module/zfs/ddt.c: In function ‘ddt_load’:
./include/sys/dmu.h:377:24: warning: ‘%s’ directive output may be truncated writing likely 5 or more bytes into a region of size between 1 and 104 [-Wformat-truncation=]
  377 | #define DMU_POOL_DDT   "DDT-%s-%s-%s"
      |                        ^~~~~~~~~~~~~~
module/zfs/ddt.c:377:37: note: in expansion of macro ‘DMU_POOL_DDT’
  377 |  (void) snprintf(name, DDT_NAMELEN, DMU_POOL_DDT,
      |                                     ^~~~~~~~~~~~
./include/sys/dmu.h:377:35: note: format string is defined here
  377 | #define DMU_POOL_DDT   "DDT-%s-%s-%s"
      |                                   ^~
./include/sys/dmu.h:377:24: note: assuming directive output of 5 bytes
  377 | #define DMU_POOL_DDT   "DDT-%s-%s-%s"
      |                        ^~~~~~~~~~~~~~
module/zfs/ddt.c:377:37: note: in expansion of macro ‘DMU_POOL_DDT’
  377 |  (void) snprintf(name, DDT_NAMELEN, DMU_POOL_DDT,
      |                                     ^~~~~~~~~~~~
module/zfs/ddt.c:377:9: note: ‘snprintf’ output 7 or more bytes (assuming 115) into a destination of size 110
  377 |  (void) snprintf(name, DDT_NAMELEN, DMU_POOL_DDT,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  378 |      zio_checksum_table[ddt->ddt_checksum].ci_name,
      |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  379 |      ddt_ops[type]->ddt_op_name, ddt_class_name[class]);
      |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Describe how to reproduce the problem
Just build on Debian 11, I assume.

### Include any warning/errors/backtraces from the system logs
See above.

**Comment by robn** [comment]
Sigh this thing. What compiler version?

**Comment by rincebrain** [comment]
Sorry, I meant to include that and forgot.

```
 $ gcc --version
gcc (Debian 10.2.1-6) 10.2.1 20210110
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by behlendorf** [comment]
Fixed by #15908
